### DISCUSSION TITLE
Modular Snap-Fit Enclosure System

### DISCUSSION THREAD

**Discussion by probonopd** [probonopd/MiniDexed#323] [Ideas]
Been experimenting with a modular system to hold the various components like display, keys and/or encoder, and to make them easily swappable. Everything snaps together with that super satisfying click, no screws and no glue are used anywhere. I am thinking about releasing the specs for this system, so that everyone can make compatible modules that just click into place. What do you think, would you be interested in using something like this?

![modular1](https://user-images.githubusercontent.com/2480569/180610949-2f792d75-fc84-4a1e-a1e4-aeb4b404685e.jpg)

![modular2](https://user-images.githubusercontent.com/2480569/180610955-83a96696-0136-4f2f-8ecc-74ac40daa16e.jpg)

![modular3](https://user-images.githubusercontent.com/2480569/180610956-f78db1c0-0dec-40ea-a7b9-3843676bbecb.jpg)

![modular4](https://user-images.githubusercontent.com/2480569/180610959-98834cee-afaf-4e68-84c8-bfd694dd0b90.jpg)

**Comment by dcoredump** [comment]
That's pretty cool! Currently my Ender-3 has some problems but when I can find some time to fix it, it would be very nice to print such case.

What is this fancy arrow-button-keyboard?

**Reply by probonopd** [reply]
While I am happy with how v1 turned out I am already experimenting with improvements for a v2 design.

The buttons can be [used](https://github.com/probonopd/MiniDexed/pull/274) instead of a rotary encoder, just uses 4 normal switches, again all mounted by just snapping it together. And if it turns out I don't like it, I can swap the keys module for a rotary encoder module which I have also built.

**Comment by antman49443** [comment]
Great design! This would be perfect for me. I rather dislike the encoder interface and prefer buttons, but I'm guessing if someone prefers the encoder, they could swap out the button module. Good thinking. I suppose a 1/8" TRS jack could be placed in the button module for traditional midi in as well.

**Reply by probonopd** [reply]
Well... where would the innards go in such a small box? But hey, it's modular. Who says we can't make larger versions of this? ;-)

**Comment by probonopd** [comment]
Constructing 3D models can be such a time sink... as soon as you have something (and are pretty happy with it) you start to see ways to improve upon it... but v2 is coming along very nicely.

**Comment by positionhigh** [comment]
From my experience, snap fit is great but only for a few cycles. After that, everything gets kind of loose. I have changed to threaded inserts like this:
https://ultimaker.com/learn/heat-set-inserts

**Reply by probonopd** [reply]
Well. The key to snap fit is that you need to work with clearances, and the snaps must _not_ be under pressure all the time while in use (pressure must only be there during dis/assembly). If they are designed like that, they can work for a _long_ time.

Some people design press fit instead of snap fit connections. Those are under pressure all the time. With the result that they stop working after a few months.

**Comment by positionhigh** [comment]
Very possible that i did non experiment with it long enough, however i printed and tested like 20-30 iterations :_)  Let me know when you have a working ruleset/construction kit for your system, i will gladly look into it.

**Reply by probonopd** [reply]
Here are my preliminary [FDM 3D Printing Design Rules for Snap Fit Cases](https://gist.github.com/probonopd/42d326dde1f7b53b3066279ddf80fff4).

**Reply by positionhigh** [reply]
Thanks for sharing that, i will definitly look into your work!

**Comment by probonopd** [comment]
Similar concept:

https://www.heise.de/news/Jetzt-in-der-Make-3-24-Modulare-Frontplatten-9749215.html

![image](https://github.com/probonopd/MiniDexed/assets/2480569/793e59ac-32c8-4019-b768-2c0317ddfd17)

but they are using screws while my objective is to do without.
